### PR TITLE
Fix gas estimation

### PR DESCRIPTION
**Pull request by n-hutton**
The gas estimation was pretty broken such that the amount it estimated was based on the gas limit and not gas used.

This was caused by the 'estimate' flag used when passed to the evm. There might be a bug in the EVM but what happens is that when estimating the gas mde during a contract call it creates a 'gas stipend' of 95% of available gas, for the new stack to use (detailed here: https://leftasexercise.com/2021/09/19/q-understanding-the-ethereum-virtual-machine-part-iii/ ) - this is logged as a 'gas cost' to the gasometer and it should recieve a refund of the whole stipend after the contract call, but what in fact happens is in the gas estimate case is this is never reverted. So, the gas used after execution is always 95% of the gas limit!

Since the external behaviour/state writes are the same irrespective of the flag, I suggest we disable it.

There are other quality of life debugging changes. Comments inline. 

Update: the gas estimation was fixed in this PR: https://github.com/Zilliqa/Zilliqa/pull/3228

### CODE REVIEW

File: src/cmd/isolated_server.cpp [hunk 1]
```diff
@@ -99,7 +99,12 @@ int main(int argc, const char* argv[]) {
   bool nonisoload{false};
   string uuid;
 
+  ENABLE_EVM = true;
```
Comment (n-hutton) [inline src/cmd/isolated_server.cpp:102]: So you do not have to change constants.xml every time.

File: src/libData/AccountData/AccountStoreSCEvm.cpp [hunk 2]
```diff
@@ -236,7 +236,7 @@ bool AccountStoreSC<MAP>::UpdateAccountsEvm(
   error_code = TxnStatus::NOT_PRESENT;
   const Address fromAddr = transaction.GetSenderAddr();
 
-  uint64_t gasRemained = transaction.GetGasLimitEth();
+  uint64_t gasLimitEth = transaction.GetGasLimitEth();
```
Comment (n-hutton) [inline src/libData/AccountData/AccountStoreSCEvm.cpp:239]: incorrectly named variable

File: src/libData/AccountData/AccountStoreSCEvm.cpp [hunk 3]
```diff
@@ -585,6 +611,8 @@ bool AccountStoreSC<MAP>::UpdateAccountsEvm(
         return false;
       }
 
+      // TODO: the cum gas might not be applied correctly (should be block
```
Comment (n-hutton) [inline src/libData/AccountData/AccountStoreSCEvm.cpp:614]: need to make jira ticket

File: src/libData/AccountData/AccountStoreTrie.cpp [hunk 4]
```diff
@@ -71,7 +71,6 @@ Account* AccountStoreTrie<MAP>::GetAccount(const Address& address,
     return account;
   }
 
-  LOG_GENERAL(INFO, "resetRoot:" << resetRoot);
```
Comment (n-hutton) [inline src/libData/AccountData/AccountStoreTrie.cpp:74]: logging noise

File: src/libServer/EthRpcMethods.cpp [hunk 5]
```diff
@@ -631,6 +631,7 @@ std::string EthRpcMethods::GetEthEstimateGas(const Json::Value& json) {
                              "data argument invalid");
     }
   }
+  std::transform(data.begin(), data.end(), data.begin(), ::toupper);
```
Comment (n-hutton) [inline src/libServer/EthRpcMethods.cpp:634]: so the args to evm are always uppercase (not a bugfix)

File: src/libServer/EthRpcMethods.cpp [hunk 6]
```diff
@@ -703,7 +704,7 @@ std::string EthRpcMethods::GetEthEstimateGas(const Json::Value& json) {
     throw JsonRpcException(ServerBase::RPC_INTERNAL_ERROR,
                            "Failed to get EVM call extras");
   }
-  args.set_estimate(true);
+  args.set_estimate(false);
```
Comment (n-hutton) [inline src/libServer/EthRpcMethods.cpp:707]: Bugfix here

File: tests/EvmAcceptanceTests/test/GasEstimations.js [hunk 7]
```diff
@@ -28,6 +37,16 @@ describe("Gas estimation with web3.js", function () {
...
       expect(gasAmount).to.be.at.least(CREATE2_MIN_GAS);
+
+      const result = await contract.methods
+        .installChild(123)
+        .send({gasLimit: 420000000, from: web3_helper.getPrimaryAccountAddress()});
+
+      expect(result).to.be.not.null;
+      const actualGas = result.gasUsed;
+
+      expect(actualGas).to.be.at.least(gasAmount * 0.9);
```
Comment (n-hutton) [inline tests/EvmAcceptanceTests/test/GasEstimations.js:48]: this test now makes sure the gas estimated is within 10% of what it actually costs to use

File: src/libServer/EthRpcMethods.cpp (hunk 6 above)
Comment (valeryz) [reply src/libServer/EthRpcMethods.cpp:707]: somehow this fix is already in master

File: src/libData/AccountData/AccountStoreSC.h [hunk 8]
```diff
@@ -29,7 +29,6 @@
 #include "InvokeType.h"
 #include "libUtils/DetachedFunction.h"
 #include "libUtils/Evm.pb.h"
-#include "libUtils/EvmCallParameters.h"
```
Comment (n-hutton) [inline src/libData/AccountData/AccountStoreSC.h:32]: no longer needed

**Review by valeryz — APPROVED** [review]
Thanks, useful cleanup and fixes.

**Pull request merged**